src/plugins/stake-plugins/uniswapV2/policies/VelodromeV2StakePolicy.ts

```diff
@@ -348,49 +349,56 @@ export const makeVelodromeV2StakePolicy = (options: UniswapV2LpPolicyOptions): S
 
         /*
         Staking for LP tokens:
-          1. Approve Pool Contract on LP-Contract:
-          2. Stake LP token
+          1. Check if LP staking is still being rewarded
+          2. Approve Pool Contract on LP-Contract:
+          3. Stake LP token
         */
 
-        // 1. Approve Pool Contract on LP-Contract:
-        txs.build(
-          (gasLimit =>
-            async function approveStakingPool({ signer, liquidity }) {
-              const spenderAddress = stakingContract.address
-              const allowanceResult = await lpTokenContract.allowance(signer.address, spenderAddress)
-              if (allowanceResult.gte(liquidity)) return
-
-              const approveResult = await lpTokenContract.connect(signer).approve(spenderAddress, BigNumber.from(liquidity), {
-                gasLimit,
-                gasPrice,
-                nonce: nextNonce()
-              })
-              cacheTxMetadata(approveResult.hash, parentCurrencyCode, {
-                name: metadataName,
-                category: 'Expense:Fees',
-                notes: `Approve ${metadataLpName} rewards contract`
-              })
-            })(gasLimitAcc('100000'))
-        )
-
-        // 2. Stake LP token
-        txs.build(
-          (gasLimit =>
-            async function stakeLiquidity({ signer, liquidity }) {
-              const result = await stakingContract.connect(signer)['deposit(uint256)'](liquidity, {
-                gasLimit,
-                gasPrice,
-                nonce: nextNonce()
-              })
+        const voterAddress = await stakingContract.connect(signer).voter()
+        const voterContract = new ethers.Contract(voterAddress, VELODROME_V2_VOTER, signer)
+        const isAlive = await voterContract.connect(signer).isAlive(voterAddress)
+
+        if (isAlive) {
```

Comment:
If ever in the future we report APYs for velodrome staking reward pools, we should consider making sure that no APY is reported for reward pools that are not alive on the backend.